### PR TITLE
POSTs

### PR DESCRIPTION
**Pull request by javierbustillo**

### CODE REVIEW

File: DAO/ChatDAO.py [hunk 1]
```diff
@@ -60,3 +60,9 @@ def insert_chat_group(self, chat_name, owner_id):
         cid = cursor.fetchone()[0]
         self.conn.commit()
         return cid
+
+    def insert_member(self, cid, member_to_add):
+        cursor = self.get_cursor()
+        query = "insert into chat_members (cid, uid) values (%s, %s)"
+        cursor.execute(query, (cid, member_to_add, cid, ))
```
Comment (ChristianPerez34) [inline DAO/ChatDAO.py]: Why is 
```python 
cid
```
repeated twice?

File: DAO/MessageDAO.py [hunk 2]
```diff
@@ -146,3 +145,18 @@ def insert_message(self, cid, uid, message, img):
         cursor.connection.commit()
 
         return message_id
+
+    def insert_reply(self, message, uid, mid, img=None):
+        cursor = self.get_cursor()
+        message_to_reply = self.get_message(mid)
```
Comment (ChristianPerez34) [inline DAO/MessageDAO.py]: I don't get why you are getting the message when you already have the ID

File: DAO/MessageDAO.py [hunk 3]
```diff
@@ -146,3 +145,18 @@ def insert_message(self, cid, uid, message, img):
...
+        cid = message_to_reply['cid']
```
Comment (ChristianPerez34) [inline DAO/MessageDAO.py]: Looking at this from the UI's point of view. I already know the chat group I'm posting a message to and the message I'm replying to. There's no need to extract ```cid``` from message

File: DAO/UserDAO.py [hunk 4]
```diff
@@ -62,3 +78,11 @@ def get_daily_active_users(self, date):
         cursor.execute(query, (date, end_date))
         users = cursor.fetchall()
         return users
+
+    def insert_contact(self, owner_contact, contact_uid_to_add, first_name, last_name):
+        cursor = self.get_cursor()
+        query = "insert into contacts (owner_id, contact_id, first_name, last_name) values (%s, %s, %s, %s)"
```
Comment (ChristianPerez34) [inline DAO/UserDAO.py:84]: Contacts table doesn't have ```first_name``` nor ```last_name``` fields

File: SQL Scripts/CreateDB.sql [hunk 5]
```diff
@@ -54,7 +54,10 @@ CREATE TABLE Vote(
 CREATE TABLE Contacts(
     owner_id Integer REFERENCES  Users(uid),
     contact_id Integer REFERENCES Users(uid),
+    first_name VARCHAR(30) NOT NULL,
```
Comment (ChristianPerez34) [inline SQL Scripts/CreateDB.sql:57]: Is it not redundant to have this field here and in Users table?

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

File: DAO/ChatDAO.py (hunk 1 above)
Comment (javierbustillo) [reply DAO/ChatDAO.py]: Era de una version vieja forgot to remove, fixed

File: DAO/UserDAO.py (hunk 4 above)
Comment (javierbustillo) [reply DAO/UserDAO.py:84]: Se las añadi

File: SQL Scripts/CreateDB.sql (hunk 5 above)
Comment (javierbustillo) [reply SQL Scripts/CreateDB.sql:57]: Nope, tu puedes ponerle el first name que tu quieras.

File: DAO/MessageDAO.py (hunk 3 above)
Comment (javierbustillo) [reply DAO/MessageDAO.py]: Right, no se me ocurrio mandar el cid del frontend

File: DAO/MessageDAO.py [hunk 6]
```diff
@@ -146,3 +151,16 @@ def insert_message(self, cid, uid, message, img):
         cursor.connection.commit()
 
         return message_id
+
+    def insert_reply(self, message, uid, mid, cid, img=None):
```
Comment (ChristianPerez34) [inline DAO/MessageDAO.py:155]:
```python 
def insert_reply(self, message, uid, mid, cid, img=None):
        cursor = self.get_cursor()
        rid = self.insert_message(cid, uid, message, img=img)
        query = 'INSERT INTO replies (replied_to, reply) values (%s, %s)'
        cursor.execute(query, (mid, rid))
        reply_id = cursor.fetchone()['mid']
        self.conn.commit()
        return reply_id
```

```insert_message``` method already checks if ```img``` is not None and adds image accordingly

File: Handlers/Chat.py [hunk 7]
```diff
@@ -29,9 +33,15 @@ def get_chat_members(self, cid):
     def get_chat_owner(self, cid):
         return self.chatDAO.get_owner_of_chat(cid)
 
-    def insert_chat(self, chat_name, owner_id):
-        cid = self.chatDAO.insert_chat(chat_name, owner_id)
-        return cid
+    def insert_chat(self, data):
```
Comment (ChristianPerez34) [inline Handlers/Chat.py:36]: Should have parameters visible instead of a dictionary

File: Handlers/Chat.py [hunk 8]
```diff
@@ -40,23 +50,19 @@ def insert_chat_message(self, cid, uid, message, img=None):
...
-        ]
-        chat = {
-            'cid': 1,
-            'name': 'skiribops',
-            'participants': participants
-        }
-        return chat
+        return self.messageDAO.insert_message(cid, uid, message, img=img)
+
+    def add_contact_to_chat_group(self, cid, data):
```
Comment (ChristianPerez34) [inline Handlers/Chat.py:55]: Parameters should be visible, not in a dictionary

File: Handlers/Chat.py [hunk 9]
```diff
@@ -74,23 +80,21 @@ def remove_chat(self, cid):
...
-            'message': 'message',
-            'cid': 1,
-            'contact_id': 1,
-            'likes': [],
-            'dislikes': [],
-            'img': None,
-            'replies': replies
-        }
-        return message
+    def reply_chat_message(self, data, mid):
```
Comment (ChristianPerez34) [inline Handlers/Chat.py:83]: Parameters shoulld be visible, not in a dictionary

File: Handlers/Users.py [hunk 10]
```diff
@@ -29,12 +30,23 @@ def get_user_by_id(self, uid):
     def get_contacts(self, user_id):
         return self.dao.get_contacts(user_id)
 
-    def insert_contact(self, first_name, last_name, email=None, phone=None):
-        contact = {
-            'contact_id': 3,
-            'uid': 4
-        }
-        return contact
+    def insert_contact(self, data):
```
Comment (ChristianPerez34) [inline Handlers/Users.py:33]: Parameters should be visible here, not in a dictionary

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

File: DAO/MessageDAO.py (hunk 6 above)
Comment (javierbustillo) [reply DAO/MessageDAO.py:155]: Uploaded

File: resources.py [hunk 11]
```diff
@@ -29,16 +31,20 @@ def post(self):
...
-        hashed_password = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
-        uid = UserHandler().insert_user(username=username, email=email, password=hashed_password)
+        username = data['username']
+        email = data['email']
+        password = data['password']
+        first_name = data['first_name']
+        last_name = data['last_name']
+        phone_number = data['phone_number']
+
+        uid = self.handler.insert_user(username, password, first_name, last_name, email, phone_number)
```
Comment (ChristianPerez34) [inline resources.py]: Send dictionary instead of parameters to be consistent with your other methods

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

**Review by ChristianPerez34 — APPROVED** [review]

**Pull request merged**